### ISSUE TITLE
Error when regex used in `raise_error`

### ISSUE THREAD

**Issue by krisleech** [splitwise/super_diff#72]
```ruby
# $ ruby -W0 query.rb

require "bundler/inline"

gemfile(false) do
  source "https://rubygems.org"
  gem "rspec"
  gem "super_diff", '0.4.1'
  gem "pry-byebug"
end

require "super_diff/rspec"

require 'rspec/autorun'

RSpec.describe 'test' do
  it 'causes an error' do
    expect { raise('foo') }.to raise_error(/sdfsdf/)
  end
end
```

```
NoMethodError:
       undefined method `name' for /sdfsdf/:Regexp
       Did you mean?  names
     # super_diff-0.4.1/lib/super_diff/rspec/monkey_patches.rb:666:in `expected_for_matcher_text'
```

* super_diff 0.4.1
* ruby 2.7

**Comment by mcmire** [comment]
This should be fixed in v0.4.2!

**Comment by krisleech** [comment]
:heart: Much appreciated!
